packages/multichain/src/caip25Permission.ts

```diff
@@ -0,0 +1,269 @@
+import type { NetworkClientId } from '@metamask/network-controller';
+import type {
+  PermissionSpecificationBuilder,
+  EndowmentGetterParams,
+  ValidPermissionSpecification,
+  PermissionValidatorConstraint,
+  PermissionConstraint,
+} from '@metamask/permission-controller';
+import {
+  CaveatMutatorOperation,
+  PermissionType,
+} from '@metamask/permission-controller';
+import type { CaipAccountId, Json } from '@metamask/utils';
+import {
+  parseCaipAccountId,
+  type Hex,
+  type NonEmptyArray,
+} from '@metamask/utils';
+import { strict as assert } from 'assert';
+import { cloneDeep, isEqual } from 'lodash';
+
+import { assertScopesSupported } from './scope/assert';
+import { validateAndFlattenScopes } from './scope/authorization';
+import type {
+  ExternalScopeString,
+  ScopeObject,
+  ScopesObject,
+} from './scope/types';
+
+export type Caip25CaveatValue = {
+  requiredScopes: ScopesObject;
+  optionalScopes: ScopesObject;
+  sessionProperties?: Record<string, Json>;
+  isMultichainOrigin: boolean;
+};
+
+export const Caip25CaveatType = 'authorizedScopes';
+
+export const Caip25CaveatFactoryFn = (value: Caip25CaveatValue) => {
+  return {
+    type: Caip25CaveatType,
+    value,
+  };
+};
+
+export const Caip25EndowmentPermissionName = 'endowment:caip25';
+
+type Caip25EndowmentSpecification = ValidPermissionSpecification<{
+  permissionType: PermissionType.Endowment;
+  targetName: typeof Caip25EndowmentPermissionName;
+  endowmentGetter: (_options?: EndowmentGetterParams) => null;
+  validator: PermissionValidatorConstraint;
+  allowedCaveats: Readonly<NonEmptyArray<string>> | null;
+}>;
+
+/**
+ * `endowment:caip25` returns nothing atm;
+ *
+ * @param builderOptions - The specification builder options.
+ * @param builderOptions.findNetworkClientIdByChainId - The hook to find the networkClientId for a chainId.
+ * @returns The specification for the `caip25` endowment.
+ */
+const specificationBuilder: PermissionSpecificationBuilder<
+  PermissionType.Endowment,
+  // TODO: FIX THIS
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  any,
```

Comment:
I think we know this type now?

Reply:
https://github.com/MetaMask/core/pull/4784/commits/c7f510ea3bd3219ae6a3a517e00412139e479f62